### PR TITLE
docs: update readme

### PR DESCRIPTION
**Pull request by yuanbohan**
## what's included

- codecov badge
- example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
+
+# PromQL Lexer and Parser
 
 The goal of this project is to build a PromQL lexer and parser capable of
 parsing PromQL that conforms with [Prometheus Query][querying-prometheus].
 
 ## Example
 
-TODO
+TODO: update example after publish to crates.io
```
Comment (waynexia) [inline README.md:10]: What kind of example do you mean?

File: README.md [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+cargo run --example parser
+```
 
 ## PromQL compliance
 
 This crate declares compatible with [prometheus 0372e25][prom-0372e25], which is
 prometheus release 2.40 at Nov 29, 2022. Any revision on PromQL after this
 commit is not guaranteed.
 
-## Design
```
Comment (waynexia) [inline README.md:16]: 🤣

File: README.md (hunk 1 above)
Comment (yuanbohan) [reply README.md:10]: add dependency in toml

File: README.md (hunk 2 above)
Comment (yuanbohan) [reply README.md:16]: Well, I think maybe some other headers are more appropriate in Readme, to introduce how this project is done, how to find Go alternative codes to challenge this one, etc.

**Review by waynexia — APPROVED** [review]
Good job 👍

**Pull request merged**